cf/commands/domain/create_shared_domain_test.go

```diff
@@ -1,73 +1,308 @@
+				}
+			}
+			return nil
+		}
 
-		Expect(runCommand("example.com")).To(BeFalse())
 	})
 
-	It("TestShareDomainFailsWithUsage", func() {
-		runCommand()
-		Expect(ui.Outputs).To(ContainSubstrings(
-			[]string{"Incorrect Usage", "Requires an argument"},
-		))
+	Describe("Requirements", func() {
+		Context("when not provided exactly one arg", func() {
+			BeforeEach(func() {
+				flagContext.Parse("arg-1", "extra-arg")
+			})
+
+			It("fails with usage", func() {
+				Expect(func() { cmd.Requirements(factory, flagContext) }).To(Panic())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Incorrect Usage. Requires DOMAIN as an argument"},
+					[]string{"NAME"},
+					[]string{"USAGE"},
+				))
+			})
+		})
+
+		Context("when provided exactly one arg", func() {
+			BeforeEach(func() {
+				flagContext.Parse("domain-name")
+			})
+
+			It("does not fail with usage", func() {
+				Expect(ui.Outputs).NotTo(ContainSubstrings(
+					[]string{"Incorrect Usage. Requires DOMAIN as an argument"},
+					[]string{"NAME"},
+					[]string{"USAGE"},
+				))
+			})
+
+			It("returns a LoginRequirement", func() {
+				actualRequirements, err := cmd.Requirements(factory, flagContext)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).To(ContainElement(loginRequirement))
+			})
+
+			Context("when router-group flag is set", func() {
+				BeforeEach(func() {
+					flagContext.Parse("domain-name", "--router-group", "route-group-name")
+				})
+
+				It("returns a LoginRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(loginRequirement))
+				})
+
+				It("returns a RoutingApiRequirement", func() {
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(factory.NewRoutingAPIRequirementCallCount()).To(Equal(1))
+					Expect(actualRequirements).To(ContainElement(routingApiRequirement))
+				})
+
+				It("returns a MinAPIVersionRequirement", func() {
+					expectedVersion, err := semver.Make("2.36.0")
+					Expect(err).NotTo(HaveOccurred())
 
-		runCommand("example.com")
-		Expect(ui.Outputs).ToNot(ContainSubstrings(
-			[]string{"Incorrect Usage", "Requires an argument"},
-		))
+					actualRequirements, err := cmd.Requirements(factory, flagContext)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(factory.NewMinAPIVersionRequirementCallCount()).To(Equal(1))
+					feature, requiredVersion := factory.NewMinAPIVersionRequirementArgsForCall(0)
+					Expect(feature).To(Equal("Option '--router-group'"))
+					Expect(requiredVersion).To(Equal(expectedVersion))
+					Expect(actualRequirements).To(ContainElement(minAPIVersionRequirement))
+				})
+			})
+
+			Context("when router-group flag is not set", func() {
```

Comment:
This context should be removed; it's covered by the context begining on line 116.